zaza/charm_tests/ceph/tests.py

```diff
@@ -512,6 +515,22 @@ def setUpClass(cls):
         """Run class setup for running ceph low level tests."""
         super(CephRGWTest, cls).setUpClass()
 
+    def setUp(self):
+        """Run instance setup for running ceph low level tests."""
+        try:
+            zaza_model.get_application(
+                'slave-ceph-radosgw')
+            self.expected_apps = [
+                'ceph-radosgw',
+                'slave-ceph-radosgw'
+            ]
+            self.multisite = True
+        except KeyError:
+            self.expected_apps = [
+                'ceph-radosgw',
+            ]
+            self.multisite = False
```

Comment:
Is this actually an exceptional situation? Or do we expect to run non multisite tests after this and the corresponding change in the parent charm is merged?

If the latter, I'd rather see us call a method that either gets the application, or returns a Falsey value if the application doesn't exist. This is probably a critique of the underlying codebase rather than with this test, though.

Reply:
Reworked a bit to use properties rather than setup during construction of test.